internal/driver/driver.go

```diff
@@ -166,6 +166,14 @@ func (c *Conn) WriteRaw(b []byte) error {
 
 // Request sends the given request to the connection and returns the response.
 func (c *Conn) Request(ctx context.Context, header *wire.MsgHeader, body wire.MsgBody) (*wire.MsgHeader, wire.MsgBody, error) {
```

Comment:
Maybe the atomic for the next RequestID should be maintained as a part of the driver, not be the caller's responsibility.

To some degree, the driver should protect the caller from making wire protocol mistakes (but not from sending valid non-sensical commands)

Reply:
this one is done I believe